### PR TITLE
Support multiple race and gender values

### PR DESCRIPTION
**Pull request by gigxz**
Update to 2020 HUD spec changes.

### CODE REVIEW

File: db/migrate/20221005133804_add_race_gender_columns_to_client.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    add_column :project_clients, :no_single_gender, :boolean, default: false
+    add_column :project_clients, :transgender, :boolean, default: false
+    add_column :project_clients, :questioning, :boolean, default: false
+
+    add_column :clients, :female, :boolean, default: false
+    add_column :clients, :male, :boolean, default: false
+    add_column :clients, :no_single_gender, :boolean, default: false
+    add_column :clients, :transgender, :boolean, default: false
+    add_column :clients, :questioning, :boolean, default: false
+  end
```
Comment (gigxz) [inline db/migrate/20221005133804_add_race_gender_columns_to_client.rb:26]: Should I remove the old race and gender columns and drop the Race and Gender tables here as well? Should I add as a second migration? Not sure what we typically do when removing tables/columns. @eanders

File: app/views/clients/_demographics.haml [hunk 2]
```diff
@@ -6,23 +7,24 @@
...
           %span.c-card__content-block-text.col-md-12.col-lg-6.c-card__content-block-text--secondary= "(#{date_format(@client.date_of_birth)})" if @client.date_of_birth
         .col-md-12.col-lg-6.c-card__content-block
           %h5 Gender
-          %span.c-card__content-block-text= @client.gender.try(:text)
+          %span.c-card__content-block-text= find_and_preserve(@client.gender_descriptions.join(',<br>')).presence || not_specified
+
         .col-md-12.col-lg-6.c-card__content-block
           %h5 Race
-          %span.c-card__content-block-text= @client.race.try(:text)
+          %span.c-card__content-block-text= find_and_preserve(@client.race_descriptions.join(',<br>')).presence || not_specified
```
Comment (gigxz) [inline app/views/clients/_demographics.haml:18]: Looks like this. I also added "not specified" to all fields.

Screen Shot 2022-10-05 at 2 35 45 PM

Screen Shot 2022-10-05 at 2 40 08 PM

Comment (eanders) [reply app/views/clients/_demographics.haml:18]: This looks good.  Thanks

File: db/migrate/20221005133804_add_race_gender_columns_to_client.rb (hunk 1 above)
Comment (eanders) [reply db/migrate/20221005133804_add_race_gender_columns_to_client.rb:26]: yes, let's drop those columns and tables.  We'll also need to update the gender rules.

**Review by eanders — CHANGES_REQUESTED** [review]
Can you track down any usages of the gender and race columns?
Here's one:
https://github.com/greenriver/boston-cas/blob/production/app/views/match_client_details/show.html.haml
We'll need to update all of those to the new version.

**Comment by gigxz** [comment]
This has been updated @eanders and should be ready for another review.
- fixed all usages of Race/Gender
- collect multiple races/genders for identified clients, and export them in csv download
- update ProjectClient correctly with race/gender for non hmis clients
- collect multiple genders for household members in Pathways 3
- remove gender and race db tables

File: app/models/identified_client.rb [hunk 3]
```diff
@@ -133,8 +136,8 @@ def download_data
         middle_name,
         last_name,
         date_of_birth,
-        Gender.find_by(numeric: gender)&.text,
-        Race.find_by(numeric: race)&.text,
+        gender_descriptions.join(', '),
+        race_descriptions.join(', '),
```
Comment (gigxz) [inline app/models/identified_client.rb:140]: This is for the identified clients CSV download. @eanders do you think this is okay, or should we have separate columns per value? I don't believe this gets imported anywhere.

File: db/migrate/20221006175635_drop_race_gender_tables.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      t.timestamps null: false
+    end
+
+    drop_table :genders do |t|
+      t.integer :numeric
+      t.string :text
+      t.timestamps null: false
+    end
+  end
+end
```
Comment (gigxz) [inline db/migrate/20221006175635_drop_race_gender_tables.rb:21]: @eanders let me know if you want me to combine these all into one migration

File: app/models/identified_client.rb (hunk 3 above)
Comment (eanders) [reply app/models/identified_client.rb:140]: this looks good.

File: app/views/clients/_client_list.haml [hunk 5]
```diff
@@ -63,7 +63,7 @@
                   unknown
                 = client_data_quality client, :date_of_birth
             %td
-              = client.gender.try(:text)
+              = find_and_preserve(client.gender_descriptions(brief: true).join(',<br>'))
```
Comment (eanders) [inline app/views/clients/_client_list.haml:66]: huh, who knew!

File: db/migrate/20221005190528_remove_old_race_gender.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class RemoveOldRaceGender < ActiveRecord::Migration[6.0]
+  def change
+    remove_column :project_clients, :primary_race, :string
+    remove_column :project_clients, :secondary_race, :string
+    remove_column :project_clients, :gender, :integer
+    remove_column :clients, :race_id, :integer
+    remove_column :clients, :gender_id, :integer
+    remove_column :clients, :gender_other, :string
```
Comment (eanders) [inline db/migrate/20221005190528_remove_old_race_gender.rb]: I think we should leave these temporarily.  If we remove them the two deployments need to go in lock-step or the CAS sync process will start bombing out as soon as we've deployed one or the other.  If we leave them, deploy both, then, remove them, that should maintain the dependency.

File: db/migrate/20221006175635_drop_race_gender_tables.rb (hunk 4 above)
Comment (eanders) [reply db/migrate/20221006175635_drop_race_gender_tables.rb:21]: nope, multiple is fine.

**Review by eanders — CHANGES_REQUESTED** [review]
Let's remove that one migration, then I think it looks great.  Thanks!

File: db/migrate/20221005190528_remove_old_race_gender.rb (hunk 6 above)
Comment (gigxz) [reply db/migrate/20221005190528_remove_old_race_gender.rb]: Ah right, updated.

**Pull request merged**